make fsdp folder to be public

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72084

make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

zhaojuanmao committed:
make fsdp folder to be public

make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

mrshenli reviewed (commented):
looks like the old `_fsdp` wasn't removed?

mrshenli commented on docs/source/fsdp.rst:
```diff
@@ -0,0 +1,7 @@
+========================
+
+.. autoclass:: torch.distributed.fsdp.FullyShardedDataParallel
+  :members:
```
We will need an empty new line here I think, otherwise, we will get

```
docs/source/fsdp.rst
The above files do not have correct trailing newlines; please normalize them
Error: Process completed with exit code 1.

```

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

mrshenli commented on docs/source/index.rst:
```diff
@@ -61,6 +61,7 @@ Features described in this documentation are classified by release status:
    torch.distributed <distributed>
    torch.distributed.algorithms.join <distributed.algorithms.join>
    torch.distributed.elastic <distributed.elastic>
+   torch.distributed.fsdp <distributed.fsdp>
```
this line is no longer needed, let's remove

zhaojuanmao replied:
looks like this is still needed?

having error:
You added the following module(s) to the PyTorch namespace 'torch.distributed.shard, torch.distributed.shard.sharded_optim, torch.distributed.shard.sharding_spec, torch.distributed.fsdp, torch.distributed.shard.sharded_tensor, torch.distributed.shard.sharded_tensor.ops' but they have no corresponding entry in a doc .rst file. You should either make sure that the .rst file that contains the module's documentation properly contains either '.. automodule:: mod_name' (if you do not want the paragraph added by the automodule, you can simply use py:module) or make the module private (by appending an '_' at the beginning of its name.++ exit 1

mrshenli replied:
I see, then we might need to replace `<distributed.fsdp>` with `<fsdp>` and remove line 83

YoseeF27 replied:
Starting to think دونَ إجابةٍ

YoseeF27 replied:
1643751800

YoseeF27 replied:
> looks like this is still needed?
> 
> having error:
> You added the following module(s) to the PyTorch namespace 'torch.distributed.shard, torch.distributed.shard.sharded_optim, torch.distributed.shard.sharding_spec, torch.distributed.fsdp, torch.distributed.shard.sharded_tensor, torch.distributed.shard.sharded_tensor.ops' but they have no corresponding entry in a doc .rst file. You should either make sure that the .rst file that contains the module's documentation properly contains either '.. automodule:: mod_name' (if you do not want the paragraph added by the automodule, you can simply use py:module) or make the module private (by appending an '_' at the beginning of its name.++ e

mrshenli replied:
Hey @YoseeF27, could you please stop that.

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

mrshenli commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -87,29 +95,34 @@ class TrainingState_(Enum):
     ``FullyShardedDataParallel`` is commonly shorten to FSDP.
-    .. _`Xu et al.`: https://arxiv.org/abs/2004.13336
-    .. _DeepSpeed: https://www.deepspeed.ai/
+    `Xu et al.`_: https://arxiv.org/abs/2004.13336
+    `DeepSpeed`_: https://www.deepspeed.ai/
```
Not sure the new version will work or not. But it seems the old one might be working, expect that we are missing one new line before `Xu et al.`. If this still doesn't work, we can make the reference explicit `text <link>`__

<img width="660" alt="Screen Shot 2022-02-01 at 3 20 38 PM" src="https://user-images.githubusercontent.com/16999635/152044876-bf122b8c-a5ec-4f71-9599-66b857978f5a.png">

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]

mrshenli reviewed (approved):
Stamp to unblock, please wait for tests

zhaojuanmao committed:
Update on "make fsdp folder to be public"


make fsdp folder to be public

Differential Revision: [D33903417](https://our.internmc.facebook.com/intern/diff/D33903417/)

[ghstack-poisoned]
